homeassistant/components/zone/__init__.py

```diff
@@ -141,6 +149,28 @@ def async_active_zone(
     return closest
 
 
+@callback
+def async_setup_track_zone_entity_ids(hass: HomeAssistant) -> None:
+    """Set up track of entity IDs for zones."""
+    zone_entity_ids: list[str] = hass.states.async_entity_ids(DOMAIN)
+    hass.data[ZONE_ENTITY_IDS] = zone_entity_ids
+
+    @callback
+    def _async_add_zone_entity_id(event_: Event) -> None:
+        """Add zone entity ID."""
+        zone_entity_ids.append(event_.data[ATTR_ENTITY_ID])
+        zone_entity_ids.sort()
+
+    @callback
+    def _async_remove_zone_entity_id(event_: Event) -> None:
+        """Remove zone entity ID."""
+        zone_entity_ids.remove(event_.data[ATTR_ENTITY_ID])
+        zone_entity_ids.sort()
```

Comment:
Is a sort really needed, the order will not change.

Reply:
Sadly, yes it is for backwards compatibility as everything expects them to be sorted by entity id.

Reply:
I wish the sort order was by creation order but the existing design is by entity id and it would be a breaking change to adjust it

Reply:
I just realized you meant only on remove :)

Yeah it will not change on remove

Reply:
good call

removed in 8f5d630357c1b16145072260fbc19e712259d354

Reply:
Oef